hi-ml/src/health_ml/utils/reports.py

```diff
@@ -0,0 +1,165 @@
+
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import jinja2
+import matplotlib.pyplot as plt
+import pandas as pd
+
+IMAGE_KEY_HTML = "IMAGEPATHSHTML"
+TABLE_KEY_HTML = "TABLEKEYHTML"
+
+
+class HTMLReport:
+    """
+    Create an HTML report which Azure ML is able to render. Note that the method convert_to_pdf will not
+    apply bootstrap formatting. Very limited CSS properties are applied to the PDF
+    (see [here](https://xhtml2pdf.readthedocs.io/en/latest/reference.html#supported-css-properties))
+    """
+    def __init__(self, title: str = "Report", output_folder: str = "outputs"):
+        self.report_title = title
+        self.output_folder = output_folder
+
+        report_folder = Path(output_folder)  # / title.lower().replace(" ", "_")
+        report_folder.mkdir(exist_ok=True, parents=True)
+
+        self.report_folder = report_folder
+        self.report_path_html = report_folder / (title.lower().replace(" ", "_") + '.html')
+        self.report_path_pdf = report_folder / (title.lower().replace(" ", "_") + '.pdf')
+        self.report_html = ""
+        self.template = ""
+        self.template_path = self._create_template()
+
+        self.env = jinja2.Environment(
+            loader=jinja2.FileSystemLoader('/')
+        )
+        self.render_kwargs: Dict[str, Any] = {"title": title}
+
+    @staticmethod
+    def _remove_html_end(report_stream):
+        return report_stream.replace("</p>\n</div>\n</body>\n</html>", "")
```

Comment:
This html_end string is something that you are relying on heavily, this should not be repeated as a literal.